NewApp/ngsa-csharp/Ngsa.DataService/Controllers/ResultHandler.cs

```diff
@@ -67,6 +68,48 @@ public static async Task<IActionResult> Handle<T>(Task<T> task, string method, s
             }
         }
 
+        public static async Task<IActionResult> Handle2<T>(HttpContext context, Task<T> task, string method, string errorMessage, ILogger logger)
+        {
+            // log the request
+            logger.LogInformation(method);
+
+            // return exception if task is null
+            if (task == null)
+            {
+                logger.LogError("Exception:{method} task is null {context}", method, context);
+
+                return CreateResult(errorMessage, HttpStatusCode.InternalServerError);
+            }
+
+            try
+            {
+                // return an OK object result
+                return new OkObjectResult(await task.ConfigureAwait(false));
+            }
+            catch (CosmosException ce)
+            {
+                // log and return Cosmos status code
+                if (ce.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    logger.LogWarning($"CosmosNotFound:{method}");
+                }
+                else
+                {
+                    logger.LogError($"{ce}\nCosmosException:{method}:{ce.StatusCode}:{ce.ActivityId}:{ce.Message}");
+                }
+
+                return CreateResult(errorMessage, ce.StatusCode);
+            }
+            catch (Exception ex)
+            {
+                // log and return exception
+                logger.LogError($"{ex}\nException:{method}:{ex.Message}");
+
+                // return 500 error
+                return CreateResult("Internal Server Error", HttpStatusCode.InternalServerError);
+            }
+        }
+
```

Comment:
Looking at this try-catch I was thinking it might be better to set variables values that outside of the try-catch scope to be used by CreateResult then in the catch you set the vars, log the info, and add a finally() block that would use the vars to call CreateResult and return the result.

Reply:
changed based on new logging